driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -370,14 +370,17 @@ public void onFailure(Throwable t) {
                         if (t instanceof UnsupportedProtocolVersionException) {
                             cluster.manager.logUnsupportedVersionProtocol(host, ((UnsupportedProtocolVersionException) t).getUnsupportedVersion());
                             cluster.manager.triggerOnDown(host, false);
+                            future.set(false);
                         } else if (t instanceof ClusterNameMismatchException) {
                             ClusterNameMismatchException e = (ClusterNameMismatchException) t;
                             cluster.manager.logClusterNameMismatch(host, e.expectedClusterName, e.actualClusterName);
                             cluster.manager.triggerOnDown(host, false);
+                            future.set(false);
                         } else {
-                            logger.warn("Error creating pool to " + host, t);
+                            logger.error("Error creating pool to " + host, t);
```

Comment:
Ah interesting, so previously for any Error (like OOME as documented in HostConnectionPool), we wouldn't propagate?